src/feature.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  EvaluationApi,
+  EvaluationBatchRequest,
+  EvaluationBatchRequestFlagTagsOperatorEnum,
+  EvaluationEntity,
+  createConfiguration,
+  ServerConfiguration,
+} from "flagr-client";
+
+export interface FlagCallbacks {
+  [key: string]: (...args: any) => Promise<any>;
+}
+
+export interface Variant {
+  key: string | null;
+  attachment: any | null;
+}
+
+const nullVariant: Variant = {
+  key: null,
+  attachment: null,
+};
+
+export interface Config {
+  flagrUrl: string;
+  tags?: string[];
+  tagOperator?: EvaluationBatchRequestFlagTagsOperatorEnum;
+}
+
+export function createFeature(config: Config) {
+  return new Feature(config);
+}
+
+export class Feature {
+  private _context: Object = {};
+  private _evaluationResults: Map<string, Variant> = new Map();
+  private _api: EvaluationApi;
+
+  constructor(private _config: Config) {
+    this._api = new EvaluationApi(
+      createConfiguration({
+        baseServer: new ServerConfiguration<{}>(
+          `${this._config.flagrUrl}/api/v1`,
+          {}
+        ),
+      })
+    );
+  }
+
+  setContext(context: Object) {
+    this._context = context;
+    this._evaluationResults = new Map();
+  }
+
+  addContext(context: Object) {
+    this.setContext({ ...this._context, context });
+  }
+
+  async match(flag: string, matchVariant: string = "on") {
+    const callbacks = { otherwise: async () => false };
+    callbacks[matchVariant] = async () => true;
+    return await this.evaluate(flag, callbacks);
```

Comment:
Fun fact: there's no need for `await` here (and this method doesn't need to be `async`) ! If you're returning a promise, you don't need to `await` it unless you need it's value, or you're error handling

Reply:
Yeah – this was me searching for what I thought was an issue related to out-of-order execution. Turns out, it was actually just related to the fact that flags take a couple of seconds to become available